plugins/modules/vmware_all_snapshots_info.py

```diff
@@ -0,0 +1,183 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = r'''
+---
+module: vmware_all_snapshots_info
+short_description: Gathers information about all snapshots across virtual machines in a specified vmware datacenter
+description:
+- This module collects detailed information of all the snapshots or the ones that the users decide using filtering It utilizes the PyVmomi library for communication with vCenter to retrieve snapshot data
+author:
+- Fernando Mendieta (@valkiriaaquatica)
+options:
+  datacenter:
+    description:
+      - The name of the datacenter to gather snapshot information from. You can get it in the vmware UI.
+    required: true
+    type: str
+  filters:
+    description:
+      - Optional filters to apply to the snapshot data being gathered, you can apply one or more.
+      - Filters are applied based on the variable match_type specified. If match_type exact, filters require exact matches. On the other hand when match_type includes it gets the values that contain that value. 
```

Comment:
```suggestion
      - Filters are applied based on the variable match_type specified. If match_type exact, filters require exact matches. On the other hand when match_type includes it gets the values that contain that value.
```

There's still a trailing whitespace here. But I think it's the last one (although I'm not 100% sure).